### ISSUE TITLE
AssertionError - unable to searialize search trie

### ISSUE THREAD

**Issue by suricactus** [mosra/m.css#150]
Hello! I really love the theme and the search in it, the only useful theme I found with working search field, the killer feature!

However, the project I am trying to generate a documentation for grew quite a lot in the last few versions and what was possible in the past is failing misserably now. 

Whenever I follow the instructions to build, I get:
``` 
File "/home/suricactus/work/tools/m.css/documentation/_search.py", line 356, in _serialize
    assert len(self.children) < 16 and len(self.results) < 2048, 'FAIL!!! {} {}'.format(len(self.children), len(self.results))
AssertionError: FAIL!!! 22 161
```
Which happens to be:
https://github.com/mosra/m.css/blob/master/documentation/_search.py#L348-L359

I didn't dive into code, but at first glance it is either solvable by increasing the limits, or by adding an option to ignore such cases and just don't include them if the overflow.

**Comment by mosra** [comment]
Hi! Interesting, lately people seem to be pushing the search to its limits :D 

I did a bit of related work for #123, there a project was hitting the 65k symbol limit, here you seem to be hitting a similar limit elsewhere. Increasing the limit means the search data will be considerably larger even for projects that don't have that many symbols, so I need to figure out some tradeoff there.

While a change that would make it work for your case isn't too complicated, I don't have time to look into this right now. Hopefully I'll be able to in early June.

**Comment by mosra** [comment]
(Sorry for *embarrassingly* late replies, finally got a chance to get back to this project.)

This is finally fixed as of b0cf44e4ddbf42ce79a8612563e84e00e8a75808 and 0411e1812f1274bf6e49404c0dcdf649f7637076. In this particular case it should "just magically work", as the children/result count limit is now 255 and 32767, respectively. 

If the project grows even further, the doc generation [may abort with a message suggesting you to modify various configuration options](https://mcss.mosra.cz/documentation/doxygen/#search-options), which then raise the default limits from 65k symbols and 16MB files to up to 4G for both. The defaults are picked like this to keep file sizes small by default, unfortunately estimating the packing options beforehand would be overly error prone so it requires manual intervention when those are reached.
